### ISSUE TITLE
Column caption options inconsistent with subcaptions

### ISSUE THREAD

**Issue by dragonstyle** [quarto-dev/quarto-cli#6728]
### Discussed in https://github.com/quarto-dev/quarto-cli/discussions/6571

Works:

````
---
title: Sample Quarto Document
format: html
execute:
  freeze: true
---

This is a sample of a Quarto document producing plots from a for loop and rendering to HTML.

```{python}
#| label: fig-sil-viz
#| fig-cap: Silhouette score with varying number of clusters
#| fig-subcap: true
#| column: screen-inset
#| layout-nrow: 2
#| fig-cap-location: margin
from sklearn.cluster import KMeans

from yellowbrick.cluster import silhouette_visualizer
from yellowbrick.datasets import load_credit

# Load a clustering dataset
X, y = load_credit()

# Specify rows to cluster: under 40 y/o and have either graduate or university education
X = X[(X['age'] <= 40) & (X['edu'].isin([1,2]))]

# Use the quick method and immediately show the figure 
cluster_list = range(2, 8)
for k in cluster_list:
    v = silhouette_visualizer(KMeans(k, random_state=42), X, colors='yellowbrick')
    v.show()
```

````

Doesn't:

````
---
title: Sample Quarto Document
format: html
execute:
  freeze: true
fig-cap-location: margin
---

This is a sample of a Quarto document producing plots from a for loop and rendering to HTML.

```{python}
#| label: fig-sil-viz
#| fig-cap: Silhouette score with varying number of clusters
#| fig-subcap: true
#| column: screen-inset
#| layout-nrow: 2
from sklearn.cluster import KMeans

from yellowbrick.cluster import silhouette_visualizer
from yellowbrick.datasets import load_credit

# Load a clustering dataset
X, y = load_credit()

# Specify rows to cluster: under 40 y/o and have either graduate or university education
X = X[(X['age'] <= 40) & (X['edu'].isin([1,2]))]

# Use the quick method and immediately show the figure
cluster_list = range(2, 8)
for k in cluster_list:
    v = silhouette_visualizer(KMeans(k, random_state=42), X, colors='yellowbrick')
    v.show()
```

````

**Comment by cscheid** [comment]
Let's pair on this one given #6620 ...

**Comment by cscheid** [comment]
There are two issues here:

- cap-location
- column layout

On `main`, the cap-location issue appears resolved (both documents render with the captions on the margin)

However, neither document has a 2-row layout. Strangely, Quarto _is_ emitting a 2-row layout, but something in our CSS is messing up.
